lib/mox.ex

```diff
@@ -597,6 +597,16 @@ defmodule Mox do
       stub(MockWeatherAPI, :get_temp, fn _loc -> {:ok, 30} end)
 
   `stub/3` will overwrite any previous calls to `stub/3`.
+
+   > #### Warning {: .warning}
+   >
+   > `stub/3` does **not verify that a call occurred**.
+   > If the code stops calling the stubbed function
+   > (for example, after a refactor), the test will still pass.
+   >
+   > This can lead to unused stubs lingering in tests.
+   > Always use `expect/3` when you want to ensure the
+   > function is actually invoked.
```

Comment:
```suggestion
  > #### Warning {: .warning}
  >
  > `stub/3` does **not verify that a call occurred**.
  > If the code stops calling the stubbed function
  > (for example, after a refactor), the test will still pass.
  >
  > This can lead to unused stubs lingering in tests.
  > Always use `expect/3` when you want to ensure the
  > function is actually invoked.
```